core/system/inc/system.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2013-2015, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __SYSTEM_H__
+#define __SYSTEM_H__
+
+#include <uvisor.h>
+#include "unvic.h"
+
+/* Default vector table (placed in Flash) */
+extern const TIsrVector g_isr_vector[ISR_VECTORS];
+
+/* Default ISRs prototypes */
+void isr_default_sys_handler(void);
+void isr_default_handler(void);
+
+/* System IRQs */
+extern void NonMaskableInt_IRQn_Handler(void);
+extern void HardFault_IRQn_Handler(void);
+extern void MemoryManagement_IRQn_Handler(void);
+extern void BusFault_IRQn_Handler(void);
+extern void UsageFault_IRQn_Handler(void);
+extern void SVCall_IRQn_Handler(void);
+extern void DebugMonitor_IRQn_Handler(void);
+extern void PendSV_IRQn_Handler(void);
+extern void SysTick_IRQn_Handler(void);
```

Comment:
I would keep them because the reason they are not needed is a "collateral". By [aliasing them](https://github.com/ARMmbed/uvisor/pull/139/files#diff-eadeeb38ba6620c41387e36e1f53d6adR20) we declare an `extern` symbol which is then referenced in the same file. At link time the declaration is solved and no error is thrown.

But if the alias is not there the same code in `system.c` fails without the `extern`s of `system.h`. As a general practice I'd always externally declare symbols that might be referenced somewhere else.